### PR TITLE
feat: create the component migration checklist

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

This PR will add our component migration guide and checklist to this repository. Closes issue https://github.com/appsmithorg/appsmith/issues/15228

## Type of change

- This change is a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/design-system/Component-Migration-Checklist.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Steps to migrate components
+
+- [ ]  Pick the component to be migrated. Mark it as in progress (highlight in yellow) on the migration sheet [Component Migration planning](https://docs.google.com/spreadsheets/d/1pcIMvGwknhbjnG1yxKH2UnRIOPwW5lw_xiFxrJdb1E8/edit?usp=drivesdk)
+- [ ]  Create an issue for it on [appsmith/ce](http://github.com/appsmithorg/appsmith). Announce that you’re doing this on #design-system
```
Comment (rohitagarwal88) [inline packages/design-system/Component-Migration-Checklist.md:4]: add #team-tech channel

File: packages/design-system/Component-Migration-Checklist.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- [ ]  Copy the files over to the [appsmith/design-system](http://github.com/appsmithorg/design-system/) repository.
+      - [ ]  Copy any unit test files over as well if they exist.
+      - [ ]  In the files, props.theme variables become variables used from variables.css
+      - [ ]  component-specific variables and tokens go in index.css (make sure you order this alphabetically)
+      - [ ]  Add stories for every state and variation of the component
+      - [ ]  Make sure to add the exports of the components you’ve added in the src/index.ts file
+- [ ]  Create a branch to change the imports on appsmith/ce.
```
Comment (rohitagarwal88) [inline packages/design-system/Component-Migration-Checklist.md:11]: @tanvibhakta Mention checklist of EE repo.

**Review by rohitagarwal88 — CHANGES_REQUESTED** [review]

**Review by albinAppsmith — APPROVED** [review]

File: packages/design-system/Component-Migration-Checklist.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - [ ]  On this branch, delete the files for the component you’ve just migrated.
+      - [ ]  Everywhere the component was used, change the import to point to the design-system package instead.
+      - [ ] If you've migrated a component that is used a specific new places in the ee-repo, make updates there as well.
```
Comment (rohitagarwal88) [inline packages/design-system/Component-Migration-Checklist.md:20]: @tanvibhakta : Also, add a mandatory step to test locally on both ce & ee. Then, the developer will know is that breaking anything on ee. Then, he needs to create a branch on ee and do import changes.

File: packages/design-system/Component-Migration-Checklist.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - [ ]  run `yarn build` to build the package, then `yalc publish` to imitate the package being published
+      - [ ]  run `yalc add '@appsmithorg/design-system'` in the appsmith/ce repository
+      - [ ]  All the imports will now be named imports. for local testing purposes, they will come from `@appsmithorg/design-system` . Make sure to change these imports to just `design-system` when committing them.
+- [ ]  Raise a PR for the changes you’ve made in the design-system repository. Make sure you have added a changeset for the PR detailing what the PR will do.
+- [ ]  Once your changes on the design-system repository has been approved and merged to release, take the beta version number of our package under ‘current tags’ from the published npm page, and replace the version number of [this line](https://github.com/appsmithorg/appsmith/blob/8428ae506a02ec477027b82936ff003c0c53cafb/app/client/package.json#L48) with it on your branch. Test once again if the imports work as expected, then raise a PR for the changed imports in the appsmith/ce repository. [@appsmithorg/design-system](https://www.npmjs.com/package/@appsmithorg/design-system)
```
Comment (rohitagarwal88) [inline packages/design-system/Component-Migration-Checklist.md:27]: @tanvibhakta : QA has to verify on both ee & ce.

**Review by rohitagarwal88 — CHANGES_REQUESTED** [review]

**Review by albinAppsmith — APPROVED** [review]

**Review by rohitagarwal88 — APPROVED** [review]

**Pull request merged**